astropy/modeling/core.py

```diff
@@ -1858,32 +1858,25 @@ def __reduce__(cls):
 
     @property
     def submodel_names(cls):
-        if cls._submodel_names is not None:
-            return cls._submodel_names
```

Comment:
The low performance of this function comes mainly from a bug in a different place (a typo really) [here](https://github.com/astropy/astropy/pull/5741/files#diff-a619f3fef9ad65068ecd1790cebc242aL1885) fixed in #5741. Nevertheless this is a possible rewrite if it speeds up additionally.